### PR TITLE
Feature/save load gpb int annotations

### PR DESCRIPTION
**Pull request by christophernorth**
This is the annotations for the Integer Vertices we can currently save without requiring extra features in the save/load code.

### CODE REVIEW

File: keanu-project/src/main/java/io/improbable/keanu/network/ProtobufWriter.java [hunk 1]
```diff
@@ -98,11 +109,13 @@ private void saveParents(KeanuSavedBayesNet.Vertex.Builder vertexBuilder,
         return parentBuilder.build();
     }
 
+    @Override
     public void saveValue(Vertex vertex) {
         //TODO - Remove once we have a version for all Vertex data types
```
Comment (GeorgeNash) [inline keanu-project/src/main/java/io/improbable/keanu/network/ProtobufWriter.java]: What's this pesky `TODO` doing here

File: keanu-project/src/main/java/io/improbable/keanu/vertices/intgr/nonprobabilistic/operators/binary/IntegerAdditionVertex.java [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
+public class IntegerAdditionVertex extends IntegerBinaryOpVertex implements SaveableVertex {
 
     /**
      * Adds one vertex to another
      *
      * @param a a vertex to add
      * @param b a vertex to add
      */
-    public IntegerAdditionVertex(IntegerVertex a, IntegerVertex b) {
+    public IntegerAdditionVertex(@LoadParentVertex(A_NAME) IntegerVertex a, @LoadParentVertex(B_NAME) IntegerVertex b) {
```
Comment (GeorgeNash) [inline keanu-project/src/main/java/io/improbable/keanu/vertices/intgr/nonprobabilistic/operators/binary/IntegerAdditionVertex.java]: How come we have `a` and `b` for integer binary ops and `left` and `right` for double binary ops? Seems like a hang over from the olden days. Can we make these consistent while we're down there?

File: keanu-project/src/main/java/io/improbable/keanu/vertices/intgr/nonprobabilistic/operators/binary/IntegerMaxVertex.java [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
+public class IntegerMaxVertex extends IntegerBinaryOpVertex implements SaveableVertex {
 
     /**
      * Finds the maximum between two vertices
      *
      * @param left  one of the vertices to find the maximum of
      * @param right one of the vertices to find the maximum of
      */
-    public IntegerMaxVertex(IntegerVertex left, IntegerVertex right) {
+    public IntegerMaxVertex(@LoadParentVertex(A_NAME) IntegerVertex left, @LoadParentVertex(B_NAME) IntegerVertex right) {
```
Comment (GeorgeNash) [inline keanu-project/src/main/java/io/improbable/keanu/vertices/intgr/nonprobabilistic/operators/binary/IntegerMaxVertex.java]: `left` and `right` here! Let's keep this consistent with the rest of the ints

File: keanu-project/src/main/java/io/improbable/keanu/network/ProtobufWriter.java (hunk 1 above)
Comment (christophernorth) [reply keanu-project/src/main/java/io/improbable/keanu/network/ProtobufWriter.java]: It's a temp thing that will get removed as part of a final pass of the APIs (once all of this + the dot stuff is out).  It needs to be there at the moment though.

File: keanu-project/src/main/java/io/improbable/keanu/vertices/intgr/nonprobabilistic/operators/binary/IntegerAdditionVertex.java (hunk 2 above)
Comment (christophernorth) [reply keanu-project/src/main/java/io/improbable/keanu/vertices/intgr/nonprobabilistic/operators/binary/IntegerAdditionVertex.java]: Done

File: keanu-project/src/main/java/io/improbable/keanu/vertices/intgr/nonprobabilistic/operators/binary/IntegerMaxVertex.java (hunk 3 above)
Comment (christophernorth) [reply keanu-project/src/main/java/io/improbable/keanu/vertices/intgr/nonprobabilistic/operators/binary/IntegerMaxVertex.java]: Done

**Pull request merged**